project-api/project-api-test/src/test/java/io/ballerina/projects/test/PackageResolutionIntegrationTests.java

```diff
@@ -329,6 +330,217 @@ public void testCase0006(ITestContext ctx) throws IOException {
         deleteDependenciesTomlAndBuildFile(projectDirPath2);
     }
 
+    @Test(description = "An imported module has a new version with a submodule")
```

Comment:
The test description can be more informative. The purpose is to test resolution of hierarchical imports.

Reply:
Please update other cases as well